schema/deploy/queries/facility_application_by_reporting_year.sql

```diff
@@ -9,7 +9,7 @@ create or replace function ggircs_portal.facility_application_by_reporting_year(
   select * from (
     (select facility.id, organisation.id, application.id,
       organisation.operator_name, facility.facility_name, facility.facility_type, facility.bcghgid,
-      facility.swrs_facility_id, application.reporting_year
+      facility.swrs_facility_id, coalesce(application.reporting_year, _reporting_year) as reporting_year
```

Comment:
Do you need a `coalesce` here, or just `_reporting_year`? If `application.reporting_year` is not null, it should be equal to `_reporting_year` because of the join condition below, right? 

Is this also needed in the second half of the union below (use `(select reporting_year from ggircs_portal.default_displayed_reporting_year())` instead of `application.reporting_year`)?